### PR TITLE
[10.0] Merge odoo brazil wip/l10n br base

### PR DESCRIPTION
**Pull request by rvalyi**
see here for the extraction script https://github.com/OCA/l10n-brazil/pull/529#issuecomment-305256100

### CODE REVIEW

File: l10n_br_base/__manifest__.py [hunk 1]
```diff
@@ -1,41 +1,106 @@
...
-    'author': 'Akretion, Odoo Community Association (OCA)',
+    'author': 'Odoo Community Association (OCA), Akretion',
     'website': 'http://odoo-brasil.org',
     'version': '10.0.1.0.0',
     'depends': [
         'base',
+        'mail',
         'base_setup',
+        'decimal_precision',
+        'product',
```
Comment (rvalyi) [inline l10n_br_base/__manifest__.py:22]: l10n_br_base should not depend on the product module, the code related to product should go in the l10n_br_product module instead

File: l10n_br_base/__manifest__.py [hunk 2]
```diff
@@ -1,41 +1,106 @@
...
+        'views/sped_participante_cliente_view.xml',
+        'views/sped_participante_fornecedor_view.xml',
+        'views/sped_participante_vincula_partner_view.xml',
+        'views/sped_empresa_view.xml',
+        'views/sped_empresa_vincula_company_view.xml',
+
+        'views/sped_produto_produto_view.xml',
+        'views/sped_unidade_produto_view.xml',
+        'views/sped_produto_servico_view.xml',
+        'views/sped_unidade_servico_view.xml',
```
Comment (rvalyi) [inline l10n_br_base/__manifest__.py:77]: l10n_br_base should not depend on the product module, the code related to product should go in the l10n_br_product module instead

File: l10n_br_base/models/__init__.py [hunk 3]
```diff
@@ -1,10 +1,60 @@
...
+#
+from . import sped_unidade
+from . import inherited_product_uom
+
+#
+# Produtos e serviços
+#
+from . import sped_produto
+from . import inherited_product_template
+from . import inherited_product_product
```
Comment (rvalyi) [inline l10n_br_base/models/__init__.py:60]: l10n_br_base should not depend on the product module, the code related to product should go in the l10n_br_product module instead

**Review by rvalyi — COMMENTED** [review]
l10n_br_base should not depend on the product module, the code related to product should go in the l10n_br_product or l10n_br_account_product module instead

copy_views dependence module should make it in OCA/server-tools instead. I'm analyzing the concept though...

File: l10n_br_base/models/inherited_mail_template.py [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+
+from odoo import api, fields, models, tools, _
+from odoo.exceptions import UserError, ValidationError
+from odoo.addons.mail.models.mail_template import (
+    mako_safe_template_env,
+    mako_template_env,
+    format_date,
+    format_tz,
+)
+from ..constante_tributaria import *
```
Comment (rvalyi) [inline l10n_br_base/models/inherited_mail_template.py:21]: this is for the SPED or product based NFE, this should hence go to the l10n_br_account_product module instead

File: l10n_br_base/models/__init__.py [hunk 5]
```diff
@@ -1,10 +1,60 @@
...
+from . import inherited_res_company
+from . import inherited_res_partner
+from . import sped_empresa
+
+#
+# Parcelamentos e pagamentos; bancos e contas bancárias
+#
+from . import copied_account_payment_term
+from . import copied_account_payment_term_line
+from . import inherited_account_payment_term
```
Comment (rvalyi) [inline l10n_br_base/models/__init__.py:47]: this code is dependent on the account module, it should go in the l10n_br_account module or even more terminal modules instead

File: l10n_br_base/models/__init__.py [hunk 6]
```diff
@@ -1,10 +1,60 @@
...
+
+#
+# Unidade é requisito para NCM e produtos
+#
+from . import sped_unidade
+from . import inherited_product_uom
```
Comment (rvalyi) [inline l10n_br_base/models/__init__.py:53]: again, this code is dependent on the product module and shoudl go to the l10n_br_product module instead at least

**Review by rvalyi — COMMENTED** [review]
more code that should be part of the l10n_br_base module

**Comment by rvalyi** [comment]
Hello contributors: so aside from my previous comments, I also identified the messy commit who lost the original files history and fuck the merge-ability of your WIP branch into the OCA:

First here is a list of the commits around the problem:

```
commit 9685798bc47b6168d80a5d8042070b75c60faa48
Author: Ari Caldeira <aristides.caldeira@tauga.com.br>
Date:   Mon Apr 10 15:31:12 2017 -0300

    [FEA] Movendo mais models para o base
rvalyi@rvalyi-laptop:~/DEV/OCA/l10n-brazil/wip-10$ git log | grep 6aef1aa1c9cdc4eb1445f5a016763f29f159aab7 -A20 -B20
    [FIX] Mais pylint

commit 3cf6e8d91ccbea4c2b87fac3c9738ebc2adac7b3
Author: Ari Caldeira <aristides.caldeira@tauga.com.br>
Date:   Sat Apr 22 10:49:24 2017 -0300

    [FIX] Mais pylint

commit 2f2385d8dfe3ca82598d557d3e70ef7f9aeb15ee
Author: Ari Caldeira <aristides.caldeira@tauga.com.br>
Date:   Sat Apr 22 10:47:53 2017 -0300

    [FIX] Erros e avisos pylint

commit 5c36864848ad49bed4c2f3ea47422b6390f84b6d
Author: Ari Caldeira <aristides.caldeira@tauga.com.br>
Date:   Sat Apr 22 09:14:21 2017 -0300

    [FEA] Movendo template de email para base

commit 6aef1aa1c9cdc4eb1445f5a016763f29f159aab7
Author: Ari Caldeira <aristides.caldeira@tauga.com.br>
Date:   Fri Apr 21 08:35:52 2017 -0300

    [FIX] Limpeza após rebase
    [FIX] Cópia de views agora mantém sincronia no caso de herança
    [FEA] Empresa movido para l10n_br_base
    [FEA] CNAE movido para sped

commit ae5f89c9040881bce627ad105cf330c96f2ff875
Merge: 9685798 1a35c41
Author: Ari Caldeira <aristides.caldeira@tauga.com.br>
Date:   Wed Apr 19 09:04:34 2017 -0300

    Merge branch '10.0-develop' of github.com:aricaldeira/l10n-brazil-wip into 10.0-develop

commit 9685798bc47b6168d80a5d8042070b75c60faa48
Author: Ari Caldeira <aristides.caldeira@tauga.com.br>
Date:   Mon Apr 10 15:31:12 2017 -0300

    [FEA] Movendo mais models para o base
```

**The last correct commit is commit 9685798bc47b6168d80a5d8042070b75c60faa48 "[FEA] Movendo mais models para o base" from @aricaldeira is OK:**


all OCA original files of the l10n_br_base module are here:
https://github.com/odoo-brazil/l10n-brazil-wip/tree/9685798bc47b6168d80a5d8042070b75c60faa48/l10n_br_base/models

examples of proper history being kept:
https://github.com/odoo-brazil/l10n-brazil-wip/commits/9685798bc47b6168d80a5d8042070b75c60faa48/l10n_br_base/models/res_partner.py
https://github.com/odoo-brazil/l10n-brazil-wip/commits/9685798bc47b6168d80a5d8042070b75c60faa48/l10n_br_base/models/res_company.py
https://github.com/odoo-brazil/l10n-brazil-wip/commits/9685798bc47b6168d80a5d8042070b75c60faa48/l10n_br_base/views/res_partner_view.xml




**But next commit 6aef1aa1c9cdc4eb1445f5a016763f29f159aab7 "[FIX] Limpeza após rebase" from @aricaldeira is KO:**

original files aren't here anymore:
https://github.com/odoo-brazil/l10n-brazil-wip/tree/6aef1aa1c9cdc4eb1445f5a016763f29f159aab7/l10n_br_base/models

they are moved to the backup directory:
https://github.com/odoo-brazil/l10n-brazil-wip/tree/6aef1aa1c9cdc4eb1445f5a016763f29f159aab7/l10n_br_base/models/backup

But in a way that lost the history:

example: https://github.com/odoo-brazil/l10n-brazil-wip/commits/6aef1aa1c9cdc4eb1445f5a016763f29f159aab7/l10n_br_base/models/backup/res_partner.py

history of view files is lost too:
https://github.com/odoo-brazil/l10n-brazil-wip/commits/6aef1aa1c9cdc4eb1445f5a016763f29f159aab7/l10n_br_base/views/backup/res_partner_view.xml



I'll now do some git-filter branch magic to restore the files into the WIP branch history and make a PR against the WIP branch to restore the OCA merge-ability. Next we will still have a lot of work to achieve the merge but at least we will be back on track again.


cc @renatonlima @mileo @mbcosta @aricaldeira

**Comment by mbcosta** [comment]
Hi all,
I did this https://docs.google.com/spreadsheets/d/1EVTWXOW6c6gkJ_JyLTZSj4BJRr5u_MdehuwZ4EhX9D4/edit#gid=0 comparative analisys of the OCA classes and fields VS the WIP branch fields to help us converging the sped_* modules inside the OCA. Any help or tips are welcome
cc @rvalyi @renatonlima @mileo @aricaldeira

File: l10n_br_base/models/inherited_res_currency.py [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    is_symbol = fields.Boolean(
+        string='Is symbol?',
+    )
+    is_index = fields.Boolean(
+        string='Is index?',
+    )
+    is_uom = fields.Boolean(
+        string='Is UOM?',
+    )
+    is_currency = fields.Boolean(
```
Comment (mbcosta) [inline l10n_br_base/models/inherited_res_currency.py:30]: I think is better extract the things not related to currency from this class

**Comment by mileo** [comment]
Olá

Lembrei de uma opção que podemos utilizar:

image

image

@aricaldeira @rvalyi

**Comment by rvalyi** [comment]
a inclusao desse modulo esta reportada na branch da versao 11

**Pull request closed without merging**